rpc/ethereum/namespaces/eth/filters/filters.go

```diff
@@ -145,36 +144,37 @@ func (f *Filter) Logs(_ context.Context) ([]*ethtypes.Log, error) {
 	from := f.criteria.FromBlock.Int64()
 	to := f.criteria.ToBlock.Int64()
 
-	blocks, err := f.backend.GetFilteredBlocks(from, to, f.bloomFilters, len(f.criteria.Addresses) > 0)
-	if err != nil {
-		return nil, err
-	}
+	for height := from; height <= to; height++ {
+		bloom, err := f.backend.BlockBloom(&height)
+		if err != nil {
+			return nil, err
+		}
 
-	for _, height := range blocks {
-		ethLogs, err := f.backend.GetLogsByNumber(types.BlockNumber(height))
+		filtered, err := f.blockLogs(height, bloom)
 		if err != nil {
-			return logs, errors.Wrapf(err, "failed to fetch block by number %d", height)
+			return nil, errors.Wrapf(err, "failed to fetch block by number %d", height)
 		}
 
-		for _, ethLog := range ethLogs {
-			filtered := FilterLogs(ethLog, f.criteria.FromBlock, f.criteria.ToBlock, f.criteria.Addresses, f.criteria.Topics)
-			logs = append(logs, filtered...)
+		// check logs limit
+		if len(logs)+len(filtered) > logLimit {
```

Comment:
If I still want to trigger an OOM, I can ensure that len(logs) > maxint/2 as well as len(filtered) > maxint/2 which will bypass this check as they'll overflow and become negative in the addition len(logs)+len(filtered). To properly fix this, you mos def want to check arithmetic by subtraction and comparisons individually against len(logs) then len(filtered)

Reply:
Since `len(logs) <= logLimit` and `len(filtered)` is bounded by `block gas limit / gas per log`, minimal gas per log is 375, a typical block gas limit is tens of millions, so it's pretty safe even for 32bit int, basically impossible for 64bit int.